tests/causal/cate_learning/test_meta_learners.py

```diff
@@ -394,3 +408,192 @@ def test_causal_s_classification_learner(
     mock_get_unique_treatments.assert_called()
     mock_fit_by_treatment.assert_called()
     mock_simulate_treatment_effect.assert_called()
+
+
+def test_simulate_t_learner_treatment_effect():
+    df = pd.DataFrame({
+        "x1": [1.3, 1.0, 1.8, -0.1],
+        "x2": [10, 4, 15, 6],
+        "treatment": ["A", "B", "A", "control"],
+        "target": [0, 0, 0, 1],
+    })
+
+    treatments = ["A", "B"]
+    control_name = "control"
+    prediction_column = "prediction"
+
+    control_learner = MagicMock()
+    control_learner.side_effect = lambda _: pd.DataFrame({"prediction": [1, 2, 3, 4]})
+
+    def treatment_learner_side_effect(_):
+        return pd.DataFrame({"prediction": [3, 2, 4, 4]})
+
+    treatment_learner = MagicMock()
+    treatment_learner.side_effect = treatment_learner_side_effect
+
+    learners = {
+        "control": control_learner,
+        "A": treatment_learner,
+        "B": treatment_learner,
+    }
+
+    result = _simulate_t_learner_treatment_effect(
+        df,
+        learners,
+        treatments,
+        control_name,
+        prediction_column,
+    )
+
+    expected = pd.DataFrame({
+        "x1": [1.3, 1.0, 1.8, -0.1],
+        "x2": [10, 4, 15, 6],
+        "treatment": ["A", "B", "A", "control"],
+        "target": [0, 0, 0, 1],
+        "treatment_A__uplift": [2, 0, 1, 0],
+        "treatment_B__uplift": [2, 0, 1, 0],
+        "uplift": [2, 0, 1, 0],
+        "suggested_treatment": ["treatment_A", "control", "treatment_A", "control"],
+    })
+
+    assert isinstance(result, pd.DataFrame)
+    assert_frame_equal(result, expected)
+
+
+def test_get_model_fcn():
+    # test if the fn is filtering the data
+    # test if the learner is called with the filtered data
+
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1, 0.0, 1.0, 2.2, 0.4, -5.0],
+            "x2": [10, 4, 15, 6, 5, 12, 14, 5, 12],
+            "treatment": [
+                "A",
+                "B",
+                "A",
+                "A",
+                "B",
+                "control",
+                "control",
+                "B",
+                "control",
+            ],
+            "target": [1, 1, 1, 0, 0, 1, 0, 0, 1],
+        }
+    )
```

Comment:
This data is widely repeated across the tests... we should convert it into a fixture. Something like:

```
@pytest.fixture
def <some_appropriate_name>():
    df = pd.DataFrame(
        {
            "x1": [1.3, 1.0, 1.8, -0.1, 0.0, 1.0, 2.2, 0.4, -5.0],
            "x2": [10, 4, 15, 6, 5, 12, 14, 5, 12],
            "treatment": [
                "A",
                "B",
                "A",
                "A",
                "B",
                "control",
                "control",
                "B",
                "control",
            ],
            "target": [1, 1, 1, 0, 0, 1, 0, 0, 1],
        }
    )
    return df
```